source/modules/MEI.visual.xml

```diff
@@ -1933,6 +1933,13 @@
           <rng:ref name="data.STAFFREL.basic"/>
         </datatype>
       </attDef>
+      <attDef ident="bracket.slope" usage="opt">
+         <desc xml:lang="en">Records the slope of the bracket independently of the slope of the beam (if any)
+           encompassing the events of the tuplet.</desc>
```

Comment:
It might be good to explicitly state the units (degrees presumably).

Reply:
refers to #648

Reply:
Degrees sounds good. Other thoughts, @lpugin, @rettinghaus, @kepper, others?

Reply:
The problem I have in #648, is that a non-zero degree value is not overly useful since it would only apply to one specific layout (that was specifically for beams, but also applies to tuplet brackets and brackets in general).   

A better unit would be the vertical offset of the endpoints, which remains constant over all layout spacings (more or less, sine beams may be flattened when the beam angle is too high).

In other words, perhaps something better than degrees would be in diatonic steps (MEI units).   0 means horizontal, 2 = move up by a staff line from the starting point to the ending point.

Reply:
Is this description better --

Records the slope of a bracket in degrees of deviation from horizontal. Positive values indicate upward slope, negative values measure slope downward, and a zero value indicates the bracket is horizontal.  The value here is independent of the slope of the beam (if any) encompassing the events of the tuplet.

Reply:
Degrees is problematic, because the angle depends on the spacing of the music (as seen in #648). Also you probably would want to limit it then (-90 to 90)? Slope itself is ∆y/∆x, and thus also dependent on spacing. We could define it in MEI as the difference of the distances between the endpoints relative to the staff lines. "3" would mean the end of the bracket is 3 vus higher than its start (at a horizontal staff), "-2.5" it would go down two and a half units.

Reply:
Ok,  how about modeling `@bracket.slope` on `@vo`:

```
<attDef ident="slope" usage="opt">
  <desc xml:lang="en">Records the slope of a bracket in terms of staff 
interline distance; that is, in units of 1/2 the distance between adjacent staff 
lines. Positive values indicate upward slope, negative values measure slope 
downward, and a zero value indicates the bracket is horizontal. The value here 
is independent of the slope of the beam (if any) encompassing the events of 
the tuplet.</desc>
  <datatype>
    <rng:ref name="data.MEASUREMENTSIGNED"/>
  </datatype>
</attDef>
```
To keep the description short, the text starting with "Positive values indicate ..." can go in the `<remarks>` section.

Reply:
Would `beam@slope` also change accordingly? Would be annoying to have two attributes with the same name and different types of values.

Reply:
> Ok, how about modeling @bracket.slope on `@vo`

That sounds good since a degree value will not be useful for dynamic layout control.

One thought is that `beam@slope` and similar could be changed to `@vo2` in a similar manner to `@tstamp` and `@tstamp2`:  `@vo2` would be the vertical offset of the endpoint compared to the starting point's vertical placement.   This would be clearer than using the name "slope" since that mathematically implies units of degrees.

Slur currently has `@startvo` and `@vo`, so perhaps change these to `@vo` and `@vo2`.  This would match `@tstamp` attribute naming system as well as to `@x`/`@x2` and `@y`/`@y2`.

Reply:
@craigsapp please stop mixing things up again. `vo` is a vertical offset, accompanied by `@startvo` and `@endvo`. 
@pe-ro yes, please change beam@slope.